### PR TITLE
WIP: Docker containers

### PR DESCRIPTION
**Pull request by mcarpenterjr**
## Description

Add support for Docker containers, this PR does/will contain a Dockerfile and matching docker-compose.yml for use in the creation and deployment of a Suite CRM container. 

It desires to be ambiguous so it can be used from the CLI to create an image or deploy a container as well as support the needs of automated deployment through a CI if needed.

At the lowest level simply running `docker build ./docker/` from the root of Suite should yield a fresh image built from the "master" branch of the official repository. 

At the highest (That I've thought of so far) using docker-compose, a user should be able to pass a target repository, branch and specify if the container should use an internal database or a separate container for the database.

No matter the level of abstraction Suite should install and set itself up, so when a container is started Suite is ready to rock.

## Motivation and Context

This breaks the dependence on third parties for the supply of containers, also will help with quick deployment of an instance.

In the running instance I am working from, gitlab i used to track any changes, but has to rsync anything done through the CI to the container that is host to suite. The container is an image from the third party bitnami (I have nothing against Bitnami, just the fact that I'm at their mercy when it comes to Suite) I've focused on the gitlab-ci to get started but this should work with any scriptable continuous integration out there. Or right from the command line.

## How To Test This

You should have both docker and docker-compose installed.

Running `docker-compose -f ./docker/docker-compose.yml up -d --force-recreate --always-recreate-deps` from the branches root will create a mariadb container and then build and deploy the docker container.

Running `docker build --force-rm -t smt/suitecrm:latest ./docker/` Will build an image that can then be used to deploy a container.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ ] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [X] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

In summary, at this point, the concept works, if docker is available the containers get created but the silent install part fails, and the DB conficuration is incomplete.

### CODE REVIEW

**Comment by connorshea** [comment]
It looks like this has a bunch of extra changes that aren't relevant, e.g. the CSS changes. Besides the addition of the Dockerfile, is any of this strictly necessary?

**Comment by mcarpenterjr** [comment]
Nope, it should be pared down some, probably should have synced my gitlab repo with GitHub before creating the Draft.

**Comment by gardner** [comment]
This would be great for local development and production deployment. Is there anything that I can do to help?

**Comment by isleshocky77** [comment]
It looks like you have some changes as part of this PR that should be broken into separate PRs to make it more manageable to test and merge in.

**Comment by pgorod** [comment]
@gil0109 note that this PR is marked as `Draft`, unfinished. So we're not considering this for review yet.

I hope @mcarpenterjr can finish it soon! Thanks

**Comment by 2xaronl** [comment]
I've just started looking into containerizing SuiteCRM and I found this pull request.  It looks like a LOT of effort was put into this years ago.  Any chance of @mcarpenterjr picking this back up or someone else jumping in?
